### PR TITLE
Consume dictionary change deletions from Core

### PR DESCRIPTION
**Pull request by leemaguire**
This PR removes the need for storing the previous dictionary transaction on the Cocoa side and consumes the logic implemented in Core instead. This in turn also bumps the core version to 11.0.1

### CODE REVIEW

File: RealmSwift/Tests/MapTests.swift [hunk 1]
```diff
@@ -27,21 +27,16 @@ fileprivate extension Map {
     }
 }
 
-class MapTests<M: RealmKeyedCollection, EM: RealmKeyedCollection>: TestCase where M.Key == String,
-                                                                                  M.Value == SwiftStringObject?,
-                                                                                  M.Element == SingleMapEntry<M.Key, M.Value>,
-                                                                                  EM.Key == String,
-                                                                                  EM.Value == EmbeddedTreeObject1?,
-                                                                                  EM.Element == SingleMapEntry<EM.Key, EM.Value> {
+class MapTests: TestCase {
```
Comment (jsflax) [inline RealmSwift/Tests/MapTests.swift:30]: Nice cleanup.

**Review by jsflax — APPROVED** [review]
You'll want to merge master again, otherwise this LGTM

**Comment by tgoyne** [comment]
This is missing changelog entries for the core changes pulled in.

**Comment by leemaguire** [comment]
@jsflax I'll merge master again in a subsequent PR

**Pull request merged**